From 9f8a3374e56413b50d89ac7ad9a87063cba48fa9 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Sun, 9 Oct 2016 18:52:52 +0800
Subject: [PATCH 1171/5242] MLK-13303-11 ARM: imx: add cpufreq support for
 i.mx6sll

commit  f4fbe442ab42d8ff2884515a72c6d24765816803 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add cpufreq support for i.MX6SLL, uses i.MX6SL's
opp table for now.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/mach-imx6sl.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/mach-imx6sl.c b/arch/arm/mach-imx/mach-imx6sl.c
index aa1f115..420a527 100644
--- a/arch/arm/mach-imx/mach-imx6sl.c
+++ b/arch/arm/mach-imx/mach-imx6sl.c
@@ -43,14 +43,14 @@ static inline void imx6sl_fec_init(void)
 
 static void __init imx6sl_init_late(void)
 {
-	/* cpufreq and cpuidle will be enabled later for i.MX6SLL */
-	if (cpu_is_imx6sll())
-		return;
-
 	/* imx6sl reuses imx6q cpufreq driver */
 	if (IS_ENABLED(CONFIG_ARM_IMX6Q_CPUFREQ))
 		platform_device_register_simple("imx6q-cpufreq", -1, NULL, 0);
 
+	/* cpuidle will be enabled later for i.MX6SLL */
+	if (cpu_is_imx6sll())
+		return;
+
 	imx6sl_cpuidle_init();
 }
 
-- 
1.7.9.5

